supertokens_python/recipe/passwordless/recipe_implementation.py

```diff
@@ -226,6 +226,20 @@ async def update_user(self, user_id: str,
             return UpdateUserEmailAlreadyExistsErrorResult()
         return UpdateUserPhoneNumberAlreadyExistsErrorResult()
 
+    async def delete_email_for_user(self, user_id: str, user_context: Dict[str, Any]) -> UpdateUserResult:
+        user = await self.get_user_by_id(user_id, user_context)
+        if user is None:
+            return UpdateUserUnknownUserIdErrorResult()
+
+        return await self.update_user(user_id, '', user.phone_number, user_context)
```

Comment:
This is wrong. You actually need to call the API here again and pass the email as a JSON `null` in the request and that will make it delete it from the core.

Same goes for phone number